src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckCodeFixProvider.cs

```diff
@@ -38,23 +39,37 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             SyntaxEditor editor, CancellationToken cancellationToken)
         {
             var declaratorLocations = new HashSet<Location>();
+            var firstTracker = new FirstTracker<StatementSyntax>(
+                s => s.GetNextStatement(),
+                s => s.IsFirstStatementInEnclosingBlock() || s.IsFirstStatementInSwitchSection());
```

Comment:
this seems unnecessarily abstract.  If you want such a type, just add it, with specific support for StatementSyntax.  If we decide in teh future we want to make this available for other features, we can consider how to design it to mke that possible.